We have 3 data sets, the first thing that we have to do is identify which rows of those data sets we will be using. The data set titled NBA_season1718_salary has player salaries for the year of 2017-2018. Because of this, our first model will focus on salaries for the that year. Then we start by looking at which players in this data set are repeated in the other two. Then we can focus on these players and the data available in the other two data sets to create our first model.

In [ ]:
# We start by importing our data sets
import pandas as pd
import numpy as np

salaries = pd.read_csv("data/NBA_season1718_salary.csv")
playerData1 = pd.read_csv("data/Players.csv")
playerData2 = pd.read_csv("data/Seasons_Stats.csv")
playerData3 = pd.read_csv("data/player_data.csv")

salaries.head()
playerData1.head()
playerData2.head()
playerData3.head()

name  year_start  year_end position height  weight  \
0       Alaa Abdelnaby        1991      1995      F-C   6-10   240.0   
1      Zaid Abdul-Aziz        1969      1978      C-F    6-9   235.0   
2  Kareem Abdul-Jabbar        1970      1989        C    7-2   225.0   
3   Mahmoud Abdul-Rauf        1991      2001        G    6-1   162.0   
4    Tariq Abdul-Wahad        1998      2003        F    6-6   223.0   

         birth_date                                college  
0     June 24, 1968                        Duke University  
1     April 7, 1946                  Iowa State University  
2    April 16, 1947  University of California, Los Angeles  
3     March 9, 1969             Louisiana State University  
4  November 3, 1974              San Jose State University

In [ ]:
# Lets get a list of the unique players in each of the data sets
uniqueSalaries = salaries["Player"].unique()
uniquePlayers1 = playerData1["Player"].unique()
uniquePlayers2 = playerData2["Player"].unique()
uniquePlayers3 = playerData3["name"].unique()

# This is a function that given two lists returns a list of the
# elements common amongst
def elemInComm(l1, l2):
  l1_ = set(l1)
  l2_ = set(l2)
  elems = list(l1_.intersection(l2_))
  return elems

commPlayer1 = elemInComm(uniqueSalaries, uniquePlayers1)
commPlayer2 = elemInComm(uniqueSalaries, uniquePlayers2)
commPlayer3 = elemInComm(uniqueSalaries, uniquePlayers3)
print(len(commPlayer1))
print(len(commPlayer2))
print(len(commPlayer3))

446
446
514


This suggests that 446 of the players whose salaries we have data on are in data set 1, 446 are in 2, and 514 are in 3. Lets see if players in common amongst data sets 1 and 2 are the same players.

In [ ]:
print(sorted(commPlayer1) == sorted(commPlayer2))

True


This means that it is exactly the same 446 players that sets 1 and 2 have in common. Now lets checj how many of these 446 are in common with the third dataset.

In [ ]:
totCommPlay = elemInComm(commPlayer1, commPlayer3)
print(len(totCommPlay))

446


This means that there are 446 players that we can analyze from the other data sets. Lets isolate these players to create 3 new csv's.

In [ ]:
cleanPlay1 = playerData1[playerData1['Player'].isin(totCommPlay)]
cleanPlay2 = playerData2[playerData2['Player'].isin(totCommPlay)]
cleanPlay3 = playerData3[playerData3['name'].isin(totCommPlay)]

cleanPlay1.to_csv("data/Players_2.csv", index = False)
cleanPlay2.to_csv("data/Seasons_Stats_2.csv", index = False)
cleanPlay3.to_csv("data/player_data_2.csv", index = False)